src/components/swapv2/LimitOrder/EditOrderModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { Trans } from '@lingui/macro'
+import { X } from 'react-feather'
+import { Flex, Text } from 'rebass'
+import styled from 'styled-components'
+
+import Modal from 'components/Modal'
+import { Z_INDEXS } from 'constants/styles'
+import { useCurrencyV2 } from 'hooks/Tokens'
+import { TransactionFlowState } from 'types'
+
+import LimitOrderForm, { Label } from './LimitOrderForm'
+import { calcInvert, calcPercentFilledOrder, calcRate, removeTrailingZero, uint256ToFraction } from './helpers'
+import { LimitOrder, LimitOrderStatus, RateInfo } from './type'
+
+const Wrapper = styled.div`
+  width: 100%;
+  padding: 20px 22px;
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+`
+
+const StyledLabel = styled(Label)`
+  margin-bottom: 0;
+`
+export default function EditOrderModal({
+  onCancelOrder,
+  onDismiss,
+  order,
+  note,
+  isOpen,
+  flowState,
+  setFlowState,
+  refreshListOrder,
+}: {
+  onCancelOrder: () => Promise<any>
+  onDismiss: () => void
+  order: LimitOrder
+  note: string
+  isOpen: boolean
+  flowState: TransactionFlowState
+  setFlowState: React.Dispatch<React.SetStateAction<TransactionFlowState>>
+  refreshListOrder: () => void
+}) {
+  const { status, makingAmount, takingAmount, makerAsset, takerAsset, filledTakingAmount, expiredAt } = order
+  const currencyIn = useCurrencyV2(makerAsset) ?? undefined
+  const currencyOut = useCurrencyV2(takerAsset) ?? undefined
+  const inputAmount = uint256ToFraction(makingAmount).toFixed(currencyIn?.decimals ?? 16)
+  const outputAmount = uint256ToFraction(takingAmount).toFixed(currencyOut?.decimals ?? 16)
+  const formatIn = inputAmount ? removeTrailingZero(inputAmount) : inputAmount
+  const formatOut = outputAmount ? removeTrailingZero(outputAmount) : outputAmount
+  const defaultExpire = new Date(expiredAt * 1000)
+  const rate = calcRate(formatIn, formatOut, currencyOut?.decimals ?? 18)
```

Comment:
We should not use fallback `decimals` for 18 anymore. We have chain that has different default decimal now. It might not being used right now but it might in the future causing hard to trace these fallback & maintain them. Edit `calcRate` so it's able to received `undefined` and return undefined.

Reply:
Make sense, update